clang/lib/Tooling/DependencyScanning/ModuleDepCollector.cpp

```diff
@@ -739,6 +748,11 @@ ModuleDepCollectorPP::handleTopLevelModule(const Module *M) {
   MDC.ScanInstance.getASTReader()->visitInputFileInfos(
       *MF, /*IncludeSystem=*/true,
       [&](const serialization::InputFileInfo &IFI, bool IsSystem) {
+        auto FullFilePath = ASTReader::ResolveImportedPath(
+            PathBuf, IFI.UnresolvedImportedFilename, MF->BaseDirectory);
+        if (MD.IsInSysroot)
+          MD.IsInSysroot = FullFilePath->starts_with(CurrSysroot);
+        PathBuf.resize_for_overwrite(256);
```

Comment:
I think it would be good to also gate the call to `ResolveImportedPath` on `IsInSysroot`. There is a bit of overhead to calling that.